src/Storages/ExternalDataSourceConfiguration.cpp

```diff
@@ -159,17 +159,30 @@ std::optional<ExternalDataSourceConfig> getExternalDataSourceConfiguration(const
     return std::nullopt;
 }
 
+static void validateConfigKeys(
+    const Poco::Util::AbstractConfiguration & dict_config, const String & config_prefix, const std::unordered_set<String> & allowed_keys)
+{
+    Poco::Util::AbstractConfiguration::Keys config_keys;
+    dict_config.keys(config_prefix, config_keys);
+    for (const auto & config_key : config_keys)
+    {
+        if (!allowed_keys.contains(config_key))
+            throw Exception(ErrorCodes::BAD_ARGUMENTS, "Unexpected key in config: {}", config_key);
```

Comment:
Maybe additional emphasize: `"Unexpected key 'xxx' in dictionary? source? config (postgresql/mongo/mysql?)"`

Reply:
hm, there is no information about dictionary source in this file, because it is shared between external dictionaries. Two options: 1) adding one more argument with information about source to pass into this function (I don't like this idea). 2) I can add try catch in dictionary source to add `addMessage` to exception and rethrow -- is it better?